MaterialLib/MPL/Properties/CreatePermeabilityMohrCoulombFailureIndexModel.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on June 5, 2020, 9:07 AM
+ */
+
+#include "CreatePermeabilityMohrCoulombFailureIndexModel.h"
+
+#include <string>
+
+#include "BaseLib/ConfigTree.h"
+#include "PermeabilityMohrCoulombFailureIndexModel.h"
+#include "ParameterLib/CoordinateSystem.h"
+#include "ParameterLib/Parameter.h"
+#include "ParameterLib/Utils.h"
+
+#include "Parameter.h"
+
+namespace MaterialPropertyLib
+{
+std::unique_ptr<Property> createPermeabilityMohrCoulombFailureIndexModel(
+    int const geometry_dimension,
+    BaseLib::ConfigTree const& config,
+    std::vector<std::unique_ptr<ParameterLib::ParameterBase>> const& parameters,
+    ParameterLib::CoordinateSystem const* const local_coordinate_system)
+{
+    if (geometry_dimension == 1)
+    {
+        OGS_FATAL(
+            "The PermeabilityMohrCoulombFailureIndexModel is only for 2D or 3D "
+            "problems");
+    }
+
+    //! \ogs_file_param{properties__property__type}
+    config.checkConfigParameter("type",
+                                "PermeabilityMohrCoulombFailureIndexModel");
+
+    // Second access for storage.
+    //! \ogs_file_param{properties__property__name}
+    auto property_name = config.peekConfigParameter<std::string>("name");
+
+    DBUG("Create PermeabilityMohrCoulombFailureIndexModel property {:s}.",
+         property_name);
+
+    std::string const& parameter_name =
+        //! \ogs_file_param{properties__property__Parameter__parameter_name}
+        config.getConfigParameter<std::string>("parameter_name");
+    auto const& parameter_k0 = ParameterLib::findParameter<double>(
+        parameter_name, parameters, 0, nullptr);
+
+    auto const kr =
+        //! \ogs_file_param{properties__property__PermeabilityMohrCoulombFailureIndexModel__kr}
+        config.getConfigParameter<double>("kr");
```

Comment:
Names of the tags should be understandable w/o looking into documentation. Use long names.

Reply:
changed.